packages/next/src/export/index.ts

```diff
@@ -529,46 +530,78 @@ export async function exportAppImpl(
+              renderOpts,
+              ampValidatorPath:
+                nextConfig.experimental.amp?.validator || undefined,
+              trailingSlash: nextConfig.trailingSlash,
+              serverRuntimeConfig,
+              subFolders,
+              buildExport: options.buildExport,
+              optimizeFonts: nextConfig.optimizeFonts as FontConfig,
+              optimizeCss: nextConfig.experimental.optimizeCss,
+              disableOptimizedLoading:
+                nextConfig.experimental.disableOptimizedLoading,
+              parentSpanId: pageExportSpan.getId(),
+              httpAgentOptions: nextConfig.httpAgentOptions,
+              debugOutput: options.debugOutput,
+              cacheMaxMemorySize: nextConfig.cacheMaxMemorySize,
+              fetchCache: true,
+              fetchCacheKeyPrefix: nextConfig.experimental.fetchCacheKeyPrefix,
+              cacheHandler: nextConfig.cacheHandler,
+              enableExperimentalReact: needsExperimentalReact(nextConfig),
+              enabledDirectories,
+            })
+          })
+
+          // If there was an error in the export, throw it immediately. In the catch block, we might retry the export,
+          // or immediately fail the build, depending on user configuration. We might also continue on and attempt other pages.
+          if (result && 'error' in result) {
+            throw new ExportError()
+          }
+
+          // If the export succeeds, break out of the retry loop
+          break
+        } catch (err) {
+          // The only error that should be caught here is an ExportError, as `exportPage` doesn't throw and instead returns an object with an `error` property.
+          // This is an overly cautious check to ensure that we don't accidentally catch an unexpected error.
+          if (!(err instanceof ExportError)) {
+            throw err
+          }
+
+          const { page } = exportPathMap[path]
+          const pageKey = page !== path ? `${page}: ${path}` : path
+          const currentCount = failedExportAttemptsByPage.get(pageKey) ?? 0
+          failedExportAttemptsByPage.set(pageKey, currentCount + 1)
+
+          // If we've reached the maximum number of attempts, and `prerenderEarlyExit` is enabled, throw an error to exit the build.
+          if (attempt >= maxAttempts - 1) {
+            Log.warn(
+              `Failed to export ${pageKey} after ${attempt + 1} attempts.`
+            )
+            if (nextConfig.experimental.prerenderEarlyExit) {
+              throw new ExportError(
+                `Export encountered an error on ${pageKey}, exiting the build.`
+              )
+            }
+          } else {
+            // Otherwise, we have more attempts to make. Wait for a second before retrying.
+            Log.info(
+              `Failed to export ${pageKey} (attempt ${attempt + 1} of ${maxAttempts}). Retrying in 1 second.`
+            )
+            await new Promise((r) => setTimeout(r, 1000))
```

Comment:
Ideally this would be a random time to avoid "waves" of retries. Maybe `Math.random() * 500`?